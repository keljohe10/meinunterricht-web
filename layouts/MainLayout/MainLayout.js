import React from 'react';
import PropTypes from 'prop-types';
import { ChakraProvider } from '@chakra-ui/react';
import Navbar from '@/layouts/components/Navbar';

import { Box } from '@chakra-ui/react';

const MainLayout = ({ children }) => {
  return (
    <ChakraProvider>
      <Navbar />
      <Box>{children}</Box>
    </ChakraProvider>
  );
};

MainLayout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MainLayout;
