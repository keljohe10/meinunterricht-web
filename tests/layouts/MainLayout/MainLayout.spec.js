/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';

import { render, cleanup } from '@/tests/testUtils';

import MainLayout from '@/layouts/MainLayout/MainLayout';

beforeEach(cleanup);

describe('MainLayout', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(
      <MainLayout>
        <nav>mock</nav>
      </MainLayout>,
    );
  });

  it('matches snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should render the navbar', () => {
    expect(wrapper.getByText('mock')).toBeInTheDocument();
  });
});
