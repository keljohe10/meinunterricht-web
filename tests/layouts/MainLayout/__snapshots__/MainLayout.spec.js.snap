// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-m7wkyr"
  >
    <div
      class="css-lygqz2"
    >
      <div
        class="css-79wky"
      >
        MeinUnterricht
      </div>
    </div>
  </div>
  <div
    class="css-0"
  >
    <nav>
      mock
    </nav>
  </div>
  <span
    class="chakra-env"
    hidden=""
  />
</DocumentFragment>
`;
